File: 프로그래머스/0/120841. 점의 위치 구하기/점의 위치 구하기.cs
using System;

public class Solution 
{
    public int solution(int[] dot) 
    {
        bool isNegativeX = (dot[0] < 0);
        bool isNegativeY = (dot[1] < 0);
        
        int answer = 0;
        if(!isNegativeX && !isNegativeY)
        {
            answer = 1;
        }
        else if(isNegativeX && !isNegativeY)
        {
            answer = 2;
        }
        else if(isNegativeX && isNegativeY)
        {
            answer = 3;
        }
        else if(!isNegativeX && isNegativeY)
        {
            answer = 4;
        }
        return answer;
    }
}

File: 프로그래머스/0/120841. 점의 위치 구하기/점의 위치 구하기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> dot) 
{
    int x = dot[0];
    int y = dot[1];
    
    if(x>0)
    {
        if(y>0)
        {
            return 1;
        }
        else if(y<0)
        {
            return 4;
        }
    }
    else if(x<0)
    {
        if(y>0)
        {
            return 2;
        }
        else if(y<0)
        {
            return 3;
        }
    }
}